### PR TITLE
Add api-methods

### PR DESCRIPTION
**Pull request by zhulduz**
Added empty necessary API-methods for a ticket 
https://github.com/crypto-browserify/browserify-aes/issues/31

### CODE REVIEW

**Comment by dcousens** [comment]
Does this mimic node? @calvinmetcalf?

**Comment by calvinmetcalf** [comment]
- setAutoPadding, is available and is a noop
- setAAD, getAuthTag & setAuthTag throw an error 
- also you are missing setAuthTag

set and get AuthTag are available on the ciphers and deciphers

**Comment by calvinmetcalf** [comment]
all the errors are of the form

`trying to get/set auth tag/aad in unsupported state`

**Comment by zhulduz** [comment]
Can you please clarify, what did you mean?  Is there something incorrect in this PR?

**Comment by calvinmetcalf** [comment]
you need to add a method `setAuthTag`, also all the methods except setAutoPadding should throw errors.

File: index.js [hunk 1]
```diff
@@ -27,6 +27,21 @@ CipherBase.prototype.update = function (data, inputEnc, outputEnc) {
   }
   return outData
 }
+
+CipherBase.prototype.setAutoPadding = function () {}
+
+CipherBase.prototype.getAuthTag = function () {
+  throw new Error('Subclasses should implement this')
```
Comment (calvinmetcalf) [inline index.js]: this should be

`trying to get auth tag in unsupported state`

File: index.js [hunk 2]
```diff
@@ -27,6 +27,21 @@ CipherBase.prototype.update = function (data, inputEnc, outputEnc) {
...
+}
+
+CipherBase.prototype.setAuthTag = function () {
+  throw new Error('Subclasses should implement this')
```
Comment (calvinmetcalf) [inline index.js]: this should be

`trying to set auth tag in unsupported state`

File: index.js [hunk 3]
```diff
@@ -27,6 +27,21 @@ CipherBase.prototype.update = function (data, inputEnc, outputEnc) {
...
+}
+
+CipherBase.prototype.setAAD = function () {
+  throw new Error('Subclasses should implement this')
```
Comment (calvinmetcalf) [inline index.js]: this should be

`trying to set aad in unsupported state`

**Comment by zhulduz** [comment]
I’ve updated the pull request. Please check this out and let me know if everything is OK. 
Thanks

**Comment by dcousens** [comment]
ACK

**Pull request merged**